### PR TITLE
update workflows to use hyperdrive-rs

### PR DESCRIPTION
**Pull request by dpaiton**

### CODE REVIEW

**Review by jalextowle — APPROVED** [review]

File: .github/workflows/build_wheels.yml [hunk 1]
```diff
@@ -46,13 +46,19 @@ jobs:
       - name: set up pip
         run: python -m pip install --upgrade pip
 
+      - name: check out hyperdrive-rs
+        uses: actions/checkout@v3
```
Comment (ryangoree) [inline .github/workflows/build_wheels.yml:50]:
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -30,19 +30,29 @@ jobs:
...
           token: ${{ secrets.GITHUB_TOKEN }}
 
+      # NOTE: This is needed to ensure that hyperdrive-wrappers builds correctly.
+      - name: install foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: check out hyperdrive-rs
+        uses: actions/checkout@v3
```
Comment (ryangoree) [inline .github/workflows/lint.yml:40]:
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -30,19 +30,29 @@ jobs:
...
+        with:
+          repository: delvtech/hyperdrive-rs
+          ref: "v1.0.3"
+          path: "./hyperdrive-rs"
+
       - name: check out hyperdrive
-        uses: actions/checkout@master
+        uses: actions/checkout@v3
```
Comment (ryangoree) [inline .github/workflows/lint.yml:47]:
```suggestion
        uses: actions/checkout@v4
```

**Review by ryangoree — APPROVED** [review]

**Comment by dpaiton** [comment]
Closing this PR in favor of moving stuff out of this repo

**Pull request closed without merging**